src/topology/algebra/ordered.lean

```diff
@@ -2198,22 +2198,185 @@ lemma tendsto_of_monotone {ι α : Type*} [preorder ι] [topological_space α]
+  {f : α → β} (hcf : continuous_on f (Ioo a b)) (hfa : tendsto f (nhds_within a (Ioi a)) (nhds l)) :
+  continuous_on (λ x, if x ∈ Ioo a b then f x else l) (Ico a b) :=
+begin
+  by_cases hab : a < b,
+  { have : ∀ x ∈ frontier (Ioo a b) ∩ Ico a b, x = a,
+    { rintros x ⟨ ⟨ _, hx₁ ⟩, hx₂ ⟩,
+      rw interior_Ioo at hx₁,
+      simp only [mem_Ioo, not_and, not_lt] at hx₁,
+      by_cases h : a = x,
+      { simp [h] },
+      { exfalso,
+        exact not_le_of_lt hx₂.2 (hx₁ $ lt_of_le_of_ne hx₂.1 h) } },
+    apply continuous_on_if,
+    repeat
+    { intros a' ha',
+      rw (show (λ (a_1 : α), a_1 ∈ Ioo a b) = Ioo a b, from rfl) at *,
+      try {rw inter_eq_self_of_subset_left Ioo_subset_Ico_self},
+      rw this a' ha',
+      simp only [left_mem_Ioo, if_false],
+      exact tendsto_const_nhds <|> simp only [hfa, hab, nhds_within_Ioo_eq_nhds_within_Ioi] },
+    { rw (show (λ (a_1 : α), a_1 ∈ Ioo a b) = Ioo a b, from rfl) at *,
+      rw inter_eq_self_of_subset_left Ioo_subset_Ico_self,
+      exact hcf },
+    { exact continuous_on_const } },
+  { rw Ico_eq_empty (le_of_not_lt hab),
+    exact continuous_on_empty _ }
+end
+
+lemma continuous_on_Ioc_extend_continuous_on_Ioo
+  {α β : Type*} [topological_space α] [linear_order α] [order_topology α] [topological_space β] {a b : α} {l : β} {f : α → β}
+  (hcf : continuous_on f (Ioo a b)) (hfb : tendsto f (nhds_within b (Iio b)) (nhds l)) :
+  continuous_on (λ x, if x ∈ Ioo a b then f x else l) (Ioc a b) :=
+begin
+  have := @continuous_on_Ico_extend_continuous_on_Ioo (order_dual α) _ _ _ _ _ _ _ _ f,
+  erw [dual_Ico, dual_Ioi, dual_Ioo] at this,
+  exact this hcf hfb
+end
+
+lemma continuous_extend_continuous_on_Ioo
+  {α β : Type*} {a b : α} {la lb : β} [topological_space α] [linear_order α] [topological_space β]
+  [order_closed_topology α] {f : α → β} (hab : a < b)
+  (hcf : continuous_on f (Ioo a b)) (hfa : tendsto f (nhds_within a (Ioi a)) (nhds la))
+  (hfb : tendsto f (nhds_within b (Iio b)) (nhds lb)) :
+  continuous (λ x, if x ≤ a then la else if b ≤ x then lb else f x) :=
+begin
+  have fact : Iio b ∩ Ioi a = Ioo a b,
+    from set.ext ( λ x, ⟨ λ ⟨ hx₁, hx₂ ⟩, ⟨ hx₂, hx₁ ⟩, λ ⟨ hx₁, hx₂ ⟩, ⟨ hx₂, hx₁ ⟩ ⟩ ),
+  have ha : ∀ x ∈ frontier (Iic a), x = a,
+  { rintros x ⟨ hx₁, hx₂ ⟩,
+    rw [closure_eq_iff_is_closed.mpr is_closed_Iic, mem_Iic, le_iff_lt_or_eq] at hx₁,
+    cases hx₁,
+    { exfalso,
+      change x ∈ Iio a at hx₁,
+      apply hx₂,
+      apply mem_of_mem_of_subset,
+      exact hx₁,
+      rw subset_interior_iff_subset_of_open,
+      exact Iio_subset_Iic_self,
+      exact is_open_Iio },
+    exact hx₁,
+    apply_instance },
+  have hb : ∀ x ∈ frontier (Ici b), x = b,
+  { rintros x ⟨ hx₁, hx₂ ⟩,
+    rw [closure_eq_iff_is_closed.mpr is_closed_Ici, mem_Ici, le_iff_lt_or_eq] at hx₁,
+    cases hx₁,
+    { exfalso,
+      change x ∈ Ioi b at hx₁,
+      apply hx₂,
+      apply mem_of_mem_of_subset,
+      exact hx₁,
```

Comment:
I haven't opened this in VSCode, but it looks like those four lines could be compressed to `exact hx₂ (mem_of_mem_of_subset hx₁)`, possibly with a couple of underscores.